assets/pickup-availability.js

```diff
@@ -11,7 +11,7 @@ if (!customElements.get('pickup-availability')) {
     }
 
     fetchAvailability(variantId) {
-      const variantSectionUrl = `${this.dataset.baseUrl}variants/${variantId}/?section_id=pickup-availability`;
+      const variantSectionUrl = `//${window.location.host}/${this.dataset.rootUrl}/variants/${variantId}/?section_id=pickup-availability`;
```

Comment:
Does the relative path work by itself?
```suggestion
      const variantSectionUrl = `/${this.dataset.rootUrl}/variants/${variantId}/?section_id=pickup-availability`;
```

Reply:
I'm far from being a JS expert, but from what I've tested locally and read in the doc that ended up producing a call to `https://fr-ca/variants/1234/....`

Reply:
I think you might have tested with two leading slashes, making it a protocol-relative path?

Ah, and that's because the root url always starts with a slash, so this would probably work:

```suggestion
      const variantSectionUrl = `${this.dataset.rootUrl}/variants/${variantId}/?section_id=pickup-availability`;
```

Reply:
Is there a `/` too many ? 
`window.location.host` returns a url without a forward slash at the end of it but `{{ routes.root_url }}` does have one at the beginning of the string it will output. So is the one you manually added necessary ? It ends up being like that: 
![](https://screenshot.click/21-11-u6fbb-1wos0.png)

Reply:
@ludoboludo After a quick chat, we decided to go the full js version which is a bit more verbose but ensure there is no console warnings
```
      let rootUrl = this.dataset.rootUrl;
      if (!rootUrl.endsWith("/")) {
        rootUrl = rootUrl + "/";
      }
      const variantSectionUrl = `${rootUrl}variants/${variantId}/?section_id=pickup-availability`;
```